### ISSUE TITLE
Testpaint failures

### ISSUE THREAD

**Issue by janisozaur** [OpenRCT2/OpenRCT2#10204]
Over past few releases, amount of testpaint failures has significantly increased, to 731.

The number of failing testpaint tests in the "old days" was 53.

d9806305b0ae71fec59357139bc2cd817fdb0f3e is the last commit known to have this number.

[d9806305b0ae71fec59357139bc2cd817fdb0f3e.log](https://github.com/OpenRCT2/OpenRCT2/files/3826359/d9806305b0ae71fec59357139bc2cd817fdb0f3e.log)

The following commits were the culprits:
```
commit 06f203a7e70d59f3a79bc2a2f19be890f0233a0a
Author: Ted John
Date:   Wed Dec 5 20:33:45 2018 +0000

    Implement surfaces, edges and stations as objects
```
06f203a7e70d59f3a79bc2a2f19be890f0233a0a 53 -> 200
[06f203a7e70d59f3a79bc2a2f19be890f0233a0a.log](https://github.com/OpenRCT2/OpenRCT2/files/3826360/06f203a7e70d59f3a79bc2a2f19be890f0233a0a.log)

```
commit c5e1abcfb1ada0340e7edfacc9a8152a02208f96
Author: Michał Janiszewski
Date:   Wed Jan 2 22:22:52 2019 +0100

    Fix implicit fallthroughs
```
c5e1abcfb1ada0340e7edfacc9a8152a02208f96 200 -> 202
[c5e1abcfb1ada0340e7edfacc9a8152a02208f96.log](https://github.com/OpenRCT2/OpenRCT2/files/3826358/c5e1abcfb1ada0340e7edfacc9a8152a02208f96.log)

```diff
--- 06f203a7e70d59f3a79bc2a2f19be890f0233a0a.log	2019-11-08 22:46:21.557950467 +0100
+++ c5e1abcfb1ada0340e7edfacc9a8152a02208f96.log	2019-11-08 22:47:33.658592687 +0100
@@ -8566,7 +8566,15 @@
 [ RUN      ] AirPoweredVerticalCoaster.Brakes
 [       OK ] AirPoweredVerticalCoaster.Brakes (0 ms)
 [ RUN      ] AirPoweredVerticalCoaster.ReverseWhoaBellySlope
-[       OK ] AirPoweredVerticalCoaster.ReverseWhoaBellySlope (0 ms)
+rct2: 0x008AFB14
+Call counts don't match (was 0, expected 3). [chainLift:0 rotation:0 direction:0 trackSequence:0]
+ Expected:
+sub_98197C(0xAA380000 | 22264, 0, 0, 32, 20, 1, height, 0, 6, height, 0)    = { 0, 0, height }, { 0, 6, height }, { 32, 20, 1 }
+sub_98199C(SCHEME_TRACK | 22240, 0, 0, 32, 20, 1, height, 0, 6, height, 0)    = { 0, 0, height }, { 0, 6, height }, { 32, 20, 1 }
+wooden_a_supports_paint_setup(0, 0, height, SCHEME_SUPPORTS)
+   Actual:
+
+[  FAILED  ] AirPoweredVerticalCoaster.ReverseWhoaBellySlope (0 ms)
 [ RUN      ] AirPoweredVerticalCoaster.ReverseWhoaBellyVertical
 [       OK ] AirPoweredVerticalCoaster.ReverseWhoaBellyVertical (0 ms)
 [ RUN      ] AirPoweredVerticalCoaster.AirThrustTopCap
@@ -8574,7 +8582,15 @@
 [ RUN      ] AirPoweredVerticalCoaster.AirThrustVerticalDown
 [       OK ] AirPoweredVerticalCoaster.AirThrustVerticalDown (0 ms)
 [ RUN      ] AirPoweredVerticalCoaster.AirThrustVerticalDownToLevel
-[       OK ] AirPoweredVerticalCoaster.AirThrustVerticalDownToLevel (0 ms)
+rct2: 0x008AFB24
+Call counts don't match (was 0, expected 3). [chainLift:0 rotation:0 direction:0 trackSequence:6]
+ Expected:
+sub_98197C(0xAA380000 | 22278, 0, 0, 32, 20, 1, height, 0, 6, height, 0)    = { 0, 0, height }, { 0, 6, height }, { 32, 20, 1 }
+sub_98199C(SCHEME_TRACK | 22252, 0, 0, 32, 20, 1, height, 0, 6, height, 0)    = { 0, 0, height }, { 0, 6, height }, { 32, 20, 1 }
+wooden_a_supports_paint_setup(0, 0, height, SCHEME_SUPPORTS)
+   Actual:
+
+[  FAILED  ] AirPoweredVerticalCoaster.AirThrustVerticalDownToLevel (0 ms)
 [----------] 20 tests from AirPoweredVerticalCoaster (0 ms total)
 [----------] 30 tests from InvertedHairpinCoaster
 [ RUN      ] InvertedHairpinCoaster.Flat
@@ -9975,8 +9991,8 @@
 
 [----------] Global test environment tear-down
 [==========] 3049 tests from 86 test cases ran. (0 ms total).
-[  PASSED  ] 2849 tests.
-[  FAILED  ] 200 tests, listed below:
+[  PASSED  ] 2847 tests.
+[  FAILED  ] 202 tests, listed below:
 [  FAILED  ] SpiralRollerCoaster.EndStation
 [  FAILED  ] SpiralRollerCoaster.BeginStation
 [  FAILED  ] SpiralRollerCoaster.MiddleStation
@@ -10151,6 +10167,8 @@
 [  FAILED  ] AirPoweredVerticalCoaster.EndStation
 [  FAILED  ] AirPoweredVerticalCoaster.BeginStation
 [  FAILED  ] AirPoweredVerticalCoaster.MiddleStation
+[  FAILED  ] AirPoweredVerticalCoaster.ReverseWhoaBellySlope
+[  FAILED  ] AirPoweredVerticalCoaster.AirThrustVerticalDownToLevel
 [  FAILED  ] InvertedHairpinCoaster.EndStation
 [  FAILED  ] InvertedHairpinCoaster.BeginStation
 [  FAILED  ] InvertedHairpinCoaster.MiddleStation
@@ -10178,4 +10196,4 @@
 [  FAILED  ] LimLaunchedRollerCoaster.90DegToInvertedFlatQuarterLoopUp
 [  FAILED  ] LimLaunchedRollerCoaster.InvertedFlatTo90DegQuarterLoopDown
 
-200 FAILED TESTS
+202 FAILED TESTS
```

```
commit bb73dd0dfb781e0653789057f948d613428ea142
Author: Gymnasiast
Date:   Sat Aug 31 13:14:19 2019 +0200

    Refactor track element to use 16 bytes
```
bb73dd0dfb781e0653789057f948d613428ea142 202 -> 731
[bb73dd0dfb781e0653789057f948d613428ea142.log](https://github.com/OpenRCT2/OpenRCT2/files/3826357/bb73dd0dfb781e0653789057f948d613428ea142.log)

I only provided one diff, as just this is already large enough. The others simply wouldn't fit.

**Comment by janisozaur** [comment]
It appears that 3b176cd6eb41301b70f8883b3f0add34a4483bd5 was made to fix #8585, but it might be that #8585 was a regression from c5e1abcfb1ada0340e7edfacc9a8152a02208f96

**Comment by Gymnasiast** [comment]
TestPaint has been removed, so closing this issue.
